src/domain/notification/entity.ts

```diff
@@ -0,0 +1,25 @@
+/* eslint-disable max-lines-per-function */
+import type { Eq } from 'fp-ts/lib/Eq'
+import type { Option } from 'fp-ts/lib/Option'
+
+export type NotificationID = string
+export type NotificationType = 'success' | 'info' | 'warning' | 'error'
+export type ActionType = 'refresh'
```

Comment:
Don't declare the action type at this point since the notification is parameterized by `T`. This is to enable the caller to determine the type of actions they wish to facilitate. The domain must remain agnostic to this.

Reply:
That makes sense, thank you for pointing that out. I have moved the ActionType definition to the Vue component that manages the actions, keeping the domain agnostic to the specific action types.